nixos/modules/system/etc/setup-etc.pl

```diff
@@ -1,82 +1,85 @@
+#! @perl@/bin/perl
+
+use feature qw(signatures);
 use strict;
-use File::Find;
-use File::Copy;
-use File::Path;
-use File::Basename;
-use File::Slurp;
+use warnings;
+
+use File::Basename qw(dirname);
+use File::Copy qw(copy);
+use File::Find qw(find);
+use File::Path qw(make_path rmtree);
+use File::Slurp qw(read_file write_file);
+use File::Spec::Functions qw(catfile);
 
-my $etc = $ARGV[0] or die;
+my $etc = $ARGV[0] or die("This script must be called with an argument which expresses the new /etc directory in the nix store");
```

Comment:
```suggestion
my $etc = $ARGV[0] or die("This script must be called with an argument which expresses the new /etc directory in the nix store\n");
```

Ending the `die()` with a newline avoids printing ilne numbers etc.

Reply:
is this needed? personally i like to see exactly where something fails

Reply:
I agree with @cheriimoya here. Fiddling with NetworkManager dispatcherScripts I made a mistake causing `setup-etc.pl` to fail copying the temporary symlink to its final target. The file path and line number helped me figuring out the issue (and the warning introduced in this PR will help some more in the future).